aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -97,7 +98,11 @@ private AppSyncRequestFactory() {}
             if (limit != null) {
                 builder.variable("limit", "Int", limit);
             }
-
+            if (!QueryPredicates.all().equals(predicate)) {
+                String modelName = ModelSchema.fromModelClass(modelClass).getName();
```

Comment:
If you're able to get it from the `ModelSchemaRegistry`, instead, that would prevent another reflection over the model class. Alternately, `modelClass.getSimpleName()` probably does the trick, too.

Reply:
:+1: I'll go with `modelClass.getSimpleName()`